shared/strings.ts

```diff
@@ -0,0 +1,56 @@
+const locales = {
+  "en-us": require("../shared/locales/en-us.json")
+}
+
+export type Locale = "en-us"
+export type AppType = "" | "neo-codap"
+
+export interface StringMap {
+  [key: string]: string|undefined
+}
+
+export interface TranslateOptions {
+  capitalize?: boolean
+  count?: number
+  plural?: string
+}
+
+export class Strings {
+  prefix: string
+  map: StringMap
+
+  constructor (locale:Locale, app:AppType = "") {
+    this.prefix = app.length > 0 ? `${app}:` : ""
+    this.map = locales[locale]
+  }
+
+  translate(key: string, options:TranslateOptions = {}) {
+    const withoutPrefix = key.toUpperCase()
+    const withPrefix = `${this.prefix}${withoutPrefix}`
+
+    let result: string|undefined
+    if (this.map.hasOwnProperty(withPrefix)) {
+      result = this.map[withPrefix]
+    }
+    if ((result === undefined) && this.map.hasOwnProperty(withoutPrefix)) {
+      result = this.map[withoutPrefix]
+    }
+    if (result === undefined) {
+      result = key
+    }
+
+    // TODO: when needed for other languages add suffixes for capitalization and counts like
+    // "ATTRIBUTE:capitalized" or "ATTRIBUTE:plural" or "ATTRIBUTE:plural3" and check them instead
+
+    if (options.capitalize) {
+      result = `${result.substr(0, 1).toUpperCase()}${result.substr(1)}`
+    }
+    if (options.hasOwnProperty("count")) {
+      if (options.count !== 1) {
+        result += (options.plural || "s")
```

Comment:
As your comment above suggests, simply adding 's' (or any suffix) only sometimes works in English and doesn't work in (m)any other languages, so this is clearly a stopgap. Even so, I'm not sure how this use of `options.plural` helps. I would have thought replacing `result` with `options.plural` would make more sense than appending, so that one could handle things like `goose` to `geese`.